UnityProject/Assets/Scripts/Players/PlayerManager.cs

```diff
@@ -1,128 +1,137 @@
-
-        public bool UpdatePlayer(PlayerDTO playerDTO)
-        {
-            GameObject playerToUpdate = GameObject.Find(PlayerId(playerDTO.id));
-            if (playerToUpdate == null)
-                return false;
-
-            PlayerController controller = playerToUpdate.GetComponent<PlayerController>();
-            if (controller == null)
-                return false;
-
-            // The controller will change the position, score and health.
-            controller.SetNextState(playerDTO);
-
-            return true;
-        }
-
-        public void OverwritePlayersState(PlayerDTO[] players) {
-            //RemoveAllPlayers();
-            foreach (PlayerDTO player in players) {
-                if (GameObject.Find(PlayerId(player.id)) == null)
-                {
-                    CreatePlayer(player);
-                } else 
-                {
-                    UpdatePlayer(player);
-                }
-            }
-        }
-
-        private void RemoveAllPlayers()
-        {
-            GameObject[] players = GameObject.FindGameObjectsWithTag(PLAYER_TAG);
-            foreach (GameObject player in players) {
-                Destroy(player);
-            }
-        }
-
-        public string PlayerId(int id)
-        {
-            return "player" + Convert.ToString(id);
-        }
-    }
+	public interface IPlayerManager
+	{
+		bool CreatePlayer (PlayerDTO playerDTO);
+
+		bool DeletePlayer (int id);
+
+		bool UpdatePlayer (PlayerDTO playerDTO);
+	}
+
+	public class PlayerManager : MonoBehaviour, IPlayerManager
+	{
+		public static String PLAYER_TAG = "Avatar";
+		Dictionary<string, GameObject> activePlayers = new Dictionary<string, GameObject> ();
+
+		public bool CreatePlayer (PlayerDTO playerDTO)
+		{
+			// It might have already been created.
+			if (activePlayers.ContainsKey (PlayerId (playerDTO.id)))
+				return true;
+
+			GameObject player = GameObject.CreatePrimitive (PrimitiveType.Sphere);
+			if (player == null)
+				return false;
+
+			player.tag = PLAYER_TAG;
+			player.name = PlayerId (playerDTO.id);
+			player.AddComponent<PlayerController> ();
+
+			activePlayers.Add (PlayerId (playerDTO.id), player);
+
+			Debug.Log ("Added player");
+
+			return true;
+		}
+
+		public bool DeletePlayer (int id)
+		{
+			if (!activePlayers.ContainsKey (PlayerId (id)))
+				return false;
+
+			Destroy (activePlayers [PlayerId (id)]);
+
+			activePlayers.Remove (PlayerId (id));
+
+			Debug.Log ("Deleted player");
+
+			return true;
+		}
+
+		public bool UpdatePlayer (PlayerDTO playerDTO)
+		{
+			GameObject playerToUpdate = activePlayers [PlayerId (playerDTO.id)];
+
+			if (playerToUpdate == null)
+				return false;
+
+			PlayerController controller = playerToUpdate.GetComponent<PlayerController> ();
+			if (controller == null)
+				return false;
+
+			// The controller will change the position, score and health.
+			controller.SetNextState (playerDTO);
+
+			Debug.Log ("Updated player");
+
+			return true;
```

Comment:
do we do anything special from the method returns here?

Reply:
Not yet but it leaves us with possibility of doing something with it. Right now we check if it's null etc then return false but don't do anything on the other end with it, yet